internal/cmd/ske/kubeconfig/login/login.go

```diff
@@ -0,0 +1,240 @@
+		return nil, err
+	}
+
+	obj, err = scheme.Scheme.ConvertToVersion(obj, clientauthenticationv1.SchemeGroupVersion)
+	if err != nil {
+		return nil, fmt.Errorf("ConvertToVersion: %w", err)
+	}
+
+	execCredential, ok := obj.(*clientauthenticationv1.ExecCredential)
+	if !ok {
+		return nil, fmt.Errorf("conversion to ExecCredential failed")
+	}
+	if execCredential == nil || execCredential.Spec.Cluster == nil {
+		return nil, fmt.Errorf("ExecCredential contains not all needed fields")
+	}
+	config := &SKEClusterConfig{}
+	err = json.Unmarshal(execCredential.Spec.Cluster.Config.Raw, config)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal: %w", err)
+	}
+
+	return &inputModel{
+		ClusterName: config.ClusterName,
+		ProjectId:   config.STACKITProjectID,
+		CacheKey:    fmt.Sprintf("ske-login-%x", sha256.Sum256([]byte(execCredential.Spec.Cluster.Server))),
+	}, nil
+}
+
+func getCachedKubeConfig(key string) *rest.Config {
+	cachedKubeconfig, err := cache.GetObject(key)
+	if err != nil {
+		return nil
+	}
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(cachedKubeconfig)
+	if err != nil {
+		return nil
+	}
+
+	return restConfig
+}
+
+func GetAndOutputKubeconfig(ctx context.Context, cmd *cobra.Command, apiClient *ske.APIClient, model *inputModel, fallbackToCache bool, cachedKubeconfig *rest.Config) error {
+	req := buildRequest(ctx, apiClient, model)
+	kubeconfigResponse, err := req.Execute()
+	if err != nil {
+		if fallbackToCache {
+			return output(cmd, model.CacheKey, cachedKubeconfig)
+		}
+		return fmt.Errorf("request kubeconfig: %w", err)
+	}
+
+	kubeconfig, err := clientcmd.RESTConfigFromKubeConfig([]byte(*kubeconfigResponse.Kubeconfig))
+	if err != nil {
+		if fallbackToCache {
+			return output(cmd, model.CacheKey, cachedKubeconfig)
+		}
+		return fmt.Errorf("parse kubeconfig: %w", err)
+	}
+	if err = cache.PutObject(model.CacheKey, []byte(*kubeconfigResponse.Kubeconfig)); err != nil {
+		if fallbackToCache {
+			return output(cmd, model.CacheKey, cachedKubeconfig)
+		}
+		return fmt.Errorf("cache kubeconfig: %w", err)
+	}
+
+	return output(cmd, model.CacheKey, kubeconfig)
+}
+
+func buildRequest(ctx context.Context, apiClient *ske.APIClient, model *inputModel) ske.ApiCreateKubeconfigRequest {
+	req := apiClient.CreateKubeconfig(ctx, model.ProjectId, model.ClusterName)
+	expirationSeconds := strconv.Itoa(expirationSeconds)
+
+	return req.CreateKubeconfigPayload(ske.CreateKubeconfigPayload{ExpirationSeconds: &expirationSeconds})
+}
+
+func output(cmd *cobra.Command, cacheKey string, kubeconfig *rest.Config) error {
+	outputExecCredential, err := parseKubeConfigToExecCredential(kubeconfig)
+	if err != nil {
+		_ = cache.DeleteObject(cacheKey)
+		return fmt.Errorf("convert to ExecCredential: %w", err)
+	}
+
+	output, err := json.Marshal(outputExecCredential)
+	if err != nil {
+		_ = cache.DeleteObject(cacheKey)
+		return fmt.Errorf("marshal ExecCredential: %w", err)
+	}
+
+	cmd.Print(string(output))
```

Comment:
```suggestion
	p.Outputf(string(output), nil)
```

`p` is `p *print.Printer` which is a command argument

Reply:
> // Print an output using Printf to the defined output (falling back to Stderr if not set).
> // If output format is set to none, it does nothing

Where does the output come from? I need stdout not stderr.

Reply:
`p.Outputf` is just a wrapper around `cmd.Printf` that is part of our `print` pkg that is used for all command output.

It prints to the defined output of the cobra command, which is `stdout`

The only difference is that it takes into account the configured output format on the CLI.

Reply:
Done! If I got it correctly, the configured output format would only apply if I would add something like:
```go
  switch model.OutputFormat {
  case print.JSONOutputFormat:
```
correct? Just want to be 100% sure that users aren't able to mess with the output format here.

Reply:
Yes, that's correct for other output formats such as `json` or `yaml`
However, if the users pass the `--output-format` to `none`, no output would be shown, because that is part of our global verbosity handling. 

I think this is not a problem, it wouldn't make sense to pass that option when using this particular command